management-interface.go

```diff
@@ -558,6 +558,65 @@ func WifiNetworkHandler(w http.ResponseWriter, r *http.Request) {
 	tmpl.ExecuteTemplate(w, "wifi-networks.html", wifiProps)
 }
 
+// Return info on the packages that are currently installed on the device.
+func getInstalledPackages() (string, error) {
+	var out []byte
+	err := error(nil)
+	if runtime.GOOS != "windows" {
+		// 'Nix.
+		out, err = exec.Command("/usr/bin/dpkg-query", "--show", "--showformat", "${Package}|${Maintainer}|${Version}|${Source}\n").Output()
+	} else {
+		return "", nil
+	}
+
+	if err != nil {
+		return "", err
+	}
+	return string(out), nil
+
+}
+
+// InstalledPackagesHandler shows the currently installed packages on the device.
+func InstalledPackagesHandler(w http.ResponseWriter, r *http.Request) {
+
+	type packagesResponse struct {
+		NumPackageRows  int
```

Comment:
why is the number of rows being returned? it's not used anywhere and can be trivially calculated by looking at `PackageDataRows`.